In [26]:
import pandas as pd
import timeit
train = pd.read_csv('/content/drive/MyDrive/Datasets/Human Activity Recognition with Smartphones/train.csv')
test = pd.read_csv('/content/drive/MyDrive/Datasets/Human Activity Recognition with Smartphones/test.csv')

In [27]:
print(train.info())
print(train.describe())
print('La taille du dataset train est:',train.shape)
train.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7352 entries, 0 to 7351
Columns: 563 entries, tBodyAcc-mean()-X to Activity
dtypes: float64(561), int64(1), object(1)
memory usage: 31.6+ MB
None
       tBodyAcc-mean()-X  tBodyAcc-mean()-Y  ...  angle(Z,gravityMean)      subject
count        7352.000000        7352.000000  ...           7352.000000  7352.000000
mean            0.274488          -0.017695  ...             -0.056515    17.413085
std             0.070261           0.040811  ...              0.279122     8.975143
min            -1.000000          -1.000000  ...             -1.000000     1.000000
25%             0.262975          -0.024863  ...             -0.143414     8.000000
50%             0.277193          -0.017219  ...              0.003181    19.000000
75%             0.288461          -0.010783  ...              0.107659    26.000000
max             1.000000           1.000000  ...              1.000000    30.000000

[8 rows x 562 columns]
La taille du dataset tra

tBodyAcc-mean()-X  tBodyAcc-mean()-Y  ...  subject  Activity
0           0.288585          -0.020294  ...        1  STANDING
1           0.278419          -0.016411  ...        1  STANDING
2           0.279653          -0.019467  ...        1  STANDING
3           0.279174          -0.026201  ...        1  STANDING
4           0.276629          -0.016570  ...        1  STANDING

[5 rows x 563 columns]

In [28]:
#On sépare notre jeu de données
X_train = train[train.columns[:-2]]
y_train = train['Activity']

X_test = test[test.columns[:-2]]
y_test = test['Activity']

In [29]:
#Selection du modele
from sklearn.ensemble import RandomForestClassifier
rfc = RandomForestClassifier(n_estimators = 500, oob_score = True)
start_time = timeit.default_timer()
#Entrainement du Modele sur les données brutes
model = rfc.fit(X_train, y_train)
elapsed1 = timeit.default_timer() - start_time

In [31]:
from sklearn.metrics import accuracy_score
# On regarde les performances de notre modele 
pred = rfc.predict(X_test)
accuracy1 = accuracy_score(y_test, pred)
print("accuracy {:.2f} time {:.2f}s".format(accuracy1, elapsed1))

accuracy 0.93 time 63.83s


In [32]:
from sklearn.feature_selection import SelectFromModel
# On selectionne les features les plus importantes  
select = SelectFromModel(rfc, prefit=True, threshold=0.003)
X_train2 = select.transform(X_train)
print(X_train2.shape)

(7352, 86)


On est passé de 562 features à 84 features.
Comparons maintenant les temps de calcul

In [33]:
rfc2 = RandomForestClassifier(n_estimators=500, oob_score=True)

start_time = timeit.default_timer()

rfc2 = rfc2.fit(X_train2, y_train)

X_test2 = select.transform(X_test)

elapsed2 = timeit.default_timer() - start_time
pred = rfc2.predict(X_test2)

accuracy2 = accuracy_score(y_test, pred)
print("accuracy avec toutes les features : {:.2f} time {:.2f}s".format(accuracy1, elapsed1))
print("accuracy avec nombre de features reduit {:.2f} time {:.2f}s".format(accuracy2, elapsed2))


accuracy toutes les features0.93 time 63.83s
accuracy nombre de features reduit 0.89 time 22.03s
